apps/cli/src/commands/sync.ts

```diff
@@ -0,0 +1,70 @@
+import type { Data } from "@webstudio-is/react-sdk";
+import { readFile, writeFile } from "node:fs/promises";
+import { cwd } from "node:process";
+import { join } from "node:path";
+import ora from "ora";
+import { loadProjectDataById } from "@webstudio-is/http-client";
+
+import type { Command } from "../args";
+import { ensureFileInPath, loadGlobalConfigFile } from "../fs-utils";
+import { LOCAL_APP_CONFIG_FILE_NAME, LOCAL_CONFIG_FILE } from "../constants";
+
+export const sync: Command = async (args) => {
+  const spinner = ora("Syncing project data").start();
+  let projectId = args.positionals[1];
+  let configFilePath = LOCAL_CONFIG_FILE;
+
+  /*
+    If projectId is null, we will check if the current directory is
+    a webstudio project. And load the projectId from it to sync the data.
+  */
+
+  if (projectId === undefined) {
+    try {
+      const content = await readFile(
+        join(cwd(), LOCAL_APP_CONFIG_FILE_NAME),
+        "utf8"
+      );
+      const build: Data = JSON.parse(content);
```

Comment:
Why local app config has type of project data?

Reply:
Because cli just saves all the data that comes from the api to the local file. Maybe the name is mis-leading, should it be changed 🤔

Reply:
It's just weird we havee the same file name for root config and webstudio-project folder. Why data is stored in the root and not in webstudio-project. And what webstudio-project even for?

Reply:
Yep, description says the opposite.

Reply:
Token is stores globally no? Local one would be committed.

Reply:
Yes, if the command runs in a existing project. It opens the local config. Checks the `projectId` and then checks for `token` from the global config. And makes a call for the latest build.

Reply:
The type says you store data in root/webstudio.json. Not in root/webstudio-project/webstudio.json.

Reply:
I am confused a bit, can you clarify the question once more please. Here is the behaviour with the data. 
When loaded for the first time, it stores in 

```js
export const LOCAL_APP_NAME = "webstudio-project";
export const LOCAL_APP_CONFIG_FILE_NAME = "build.json";
export const LOCAL_CONFIG_FILE = join(
  process.cwd(),
  LOCAL_APP_NAME,
  LOCAL_APP_CONFIG_FILE_NAME
);
```

`/root/user/...../webstudio-project/webstudio.json`

https://github.com/webstudio-is/webstudio-builder/blob/feat/sync-project-data/apps/cli/src/commands/sync.ts#L15

But when the projectId is null in the command. We assume that is being run inside a existing project and check if the json file is present and so the
https://github.com/webstudio-is/webstudio-builder/blob/feat/sync-project-data/apps/cli/src/commands/sync.ts#L25
which is `join(cwd(), LOCAL_APP_CONFIG_FILE_NAME)`
Because `cwd` here already constructs the path till `root/user/...../webstudio-project/` and so i am joining till `join(cwd(), LOCAL_APP_CONFIG_FILE_NAME)`


If there is any doubts on the token, the path can always be the same
```js
const GLOBA_APP_CONFIG = envPaths("webstudio").config;
const GLOBAL_CONFIG_FILE_NAME = "webstudio-config.json";
export const GLOBAL_CONFIG_FILE = join(
  GLOBA_APP_CONFIG,
  GLOBAL_CONFIG_FILE_NAME
);
```
that is going to be `root/...webstudio-nodejs/webstudio-config.json`. The package `env-paths` creates the `webstudio-nodejs` here

Reply:
Now I get it. So we just have inconsistent behavior.

Now the question is why sync with project and without project id write into different folders? Should user do `cd webstudio-project && cli sync && cd ../`?

Reply:
Yes, my assumption was. Once the project is closed, there shouldn't be any need for running commands from outside. So, the `sync` and `build` should run in the current folder. So, anyone can have multiple projects in the same folder. And each can work independently. I am actually looking into `webstudio-project` folder name to be the actual project name.

Reply:
Not quite safe as title may have characters not compatible with fs

Reply:
Maybe something like this?
```
$ webstudio sync FOLDER_NAME PROJECT_ID
$ webstudio sync FOLDER_NAME
```

Reply:
Ok, this suggestion sounds good. Will make all the final changes then. Thanks for clarifying 👍